vpr/src/route/rr_graph_indexed_data.cpp

```diff
@@ -264,119 +260,104 @@ static std::vector<size_t> count_rr_segment_types() {
-                                           L_rr_node_indices);
-        if (inode == -1)
-            continue;
-        cost_index = device_ctx.rr_nodes[inode].cost_index();
-        frac_num_seg = clb_dist * device_ctx.rr_indexed_data[cost_index].inv_length;
-        Tdel = frac_num_seg * device_ctx.rr_indexed_data[cost_index].T_linear
-               + frac_num_seg * frac_num_seg
-                     * device_ctx.rr_indexed_data[cost_index].T_quadratic;
-        Tdel_sum += Tdel / (float)clb_dist;
+    float delay_norm_fac = Tdel_sum / Tdel_num;
+
+    if (getEchoEnabled() && isEchoFileEnabled(E_ECHO_RR_GRAPH_INDEXED_DATA)) {
+        std::ofstream out_file;
+
+        out_file.open(getEchoFileName(E_ECHO_RR_GRAPH_INDEXED_DATA));
+        out_file << "Delay normalization factor: " << delay_norm_fac << std::endl;
+
+        out_file.close();
     }
 
-    return (Tdel_sum / (2. * nodes_per_chan));
+    return delay_norm_fac;
 }
 
-static void load_rr_indexed_data_T_values(int index_start,
-                                          int num_indices_to_load,
-                                          t_rr_type rr_type,
-                                          int nodes_per_chan,
-                                          const t_rr_node_indices& L_rr_node_indices) {
+static void load_rr_indexed_data_T_values() {
     /* Loads the average propagation times through segments of each index type   *
      * for either all CHANX segment types or all CHANY segment types.  It does   *
      * this by looking at all the segments in one channel in the middle of the   *
      * array and averaging the R, C, and Cinternal values of all segments of the *
      * same type and using them to compute average delay values for this type of *
      * segment. */
 
-    int itrack, cost_index;
-    float *C_total, *R_total;                                         /* [0..device_ctx.rr_indexed_data.size() - 1] */
-    double *switch_R_total, *switch_T_total, *switch_Cinternal_total; /* [0..device_ctx.rr_indexed_data.size() - 1] */
-    short* switches_buffered;
-    int* num_nodes_of_index; /* [0..device_ctx.rr_indexed_data.size() - 1] */
-    float Rnode, Cnode, Rsw, Tsw, Cinternalsw;
-
     auto& device_ctx = g_vpr_ctx.mutable_device();
+    auto& rr_nodes = device_ctx.rr_nodes;
+    auto& rr_indexed_data = device_ctx.rr_indexed_data;
+
+    auto fan_in_list = get_fan_in_list();
 
-    num_nodes_of_index = (int*)vtr::calloc(device_ctx.rr_indexed_data.size(), sizeof(int));
-    C_total = (float*)vtr::calloc(device_ctx.rr_indexed_data.size(), sizeof(float));
-    R_total = (float*)vtr::calloc(device_ctx.rr_indexed_data.size(), sizeof(float));
+    std::vector<int> num_nodes_of_index(rr_indexed_data.size(), 0);
+    std::vector<std::vector<float>> C_total(rr_indexed_data.size());
+    std::vector<std::vector<float>> R_total(rr_indexed_data.size());
 
     /* August 2014: Not all wire-to-wire switches connecting from some wire segment will
```

Comment:
Again, I think we're averaging / doing a median over all rr_nodes, not just some in one chosen channel. Should update the comment.